x/mongo/driver/topology/connection.go

```diff
@@ -104,8 +109,29 @@ func (c *connection) processInitializationError(err error) {
 
 	c.connectErr = ConnectionError{Wrapped: err, init: true}
 	if c.config.errorHandlingCallback != nil {
-		c.config.errorHandlingCallback(c.connectErr, c.generation)
+		c.config.errorHandlingCallback(c.connectErr, c.generation, c.desc.ServerID)
+	}
+}
+
+// setGenerationNumber sets the connection's generation number if a callback has been provided to do so in connection
+// configuration.
+func (c *connection) setGenerationNumber() {
+	if c.config.getGenerationFn != nil {
```

Comment:
Would `getGenerationFn` ever be `nil`? The `serverID => (generation, count)` map is used to determine generation regardless of whether we are in load balanced mode.

Reply:
For unit tests, it can be. I verified that removing the line in `pool.go` that sets this function causes some of the new spec tests to fail though.